src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -2709,11 +2712,20 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
     }
 
     void visit_Variable(const ASR::Variable_t &x) {
+        if ((compiler_options.interactive) && 
+            (std::strcmp(x.m_name, "_") == 0) && 
```

Comment:
Why is it so that at some place we compare with `(std::strcmp(x.m_name, "_") == 0)` and at some we use `(std::strcmp(x.m_name, compiler_options.po.global_underscore.c_str()) == 0)`? What is the significance of `compiler_options.po.global_underscore`?

Reply:
The previous code block's last expression's value is stored in `compiler_options.po.global_underscore`, which is `___main__global_stmts_<int>__`.
While generating IR for `compiler_options.po.global_underscore`, we store its hash in `global_underscore_hash`, then whenever we need a reference to `_` variable, we use `global_underscore_hash`. So `_` gets replaced with `___main__global_stmts_<int>__`.